tensorflow/python/framework/sparse_tensor.py

```diff
@@ -417,3 +420,62 @@ def is_sparse(x):
     `True` iff `x` is a `tf.SparseTensor` or `tf.compat.v1.SparseTensorValue`.
   """
   return isinstance(x, (SparseTensor, SparseTensorValue))
+
+def from_dense(dense, name=None):
+    """Convert a dense `np.array` or a dense tensor to a `tf.SparseTensor`
```

Comment:
Why can't we use the from_tensor to convert from a numpy array (since we already have conversions from numpy array to dense tensor with zero copies)?

Reply:
My understanding is that making a dense tensor first, and then using from_tensor, results in allocating a dense tensor on the GPU. Often, we'd want to avoid making a dense tensor at all because the tensor size might be too large to comfortably work on a GPU. 

Therefore, we want to skip the dense tensor step completely, and initialize a SparseTensor as the first instantiation of the data on a GPU. Does that make sense?

Reply:
You can allocate the dense tensor on the CPU though if you do tf.device(cpu): from_tensor

Reply:
I just tested this on Colab and I believe you're correct!

Thanks for reminding me of this functionality. I'm going to close the PR.